### ISSUE TITLE
Field radio with repeated id = undefined causes misbehaving

### ISSUE THREAD

**Issue by alesisjoan** [odoo/odoo#25107]
Impacted versions: 8.0

Steps to reproduce:

Have this repeated selections in model:

```
    rifampicina = fields.Selection([('S', 'Sensible'), ('R', 'Resistente'), ('N', 'No probado')], default='N',
                                   required=True)

    teicoplanina = fields.Selection([('S', 'Sensible'), ('R', 'Resistente'), ('N', 'No probado')], default='N',
                                    required=True)

    tmp_smz = fields.Selection([('S', 'Sensible'), ('R', 'Resistente'), ('N', 'No probado')], default='N',
                               required=True)
```

and the view:

```                            
<field name="rifampicina" widget="radio"/>
 <field name="teicoplanina" widget="radio"/>
<field name="tmp_smz" widget="radio"/>
```

Current behavior:

It renders the Input with a repeated id:

<input id="undefined_S">
<input id="undefined_R">
<input id="undefined_N">

It occurs that if you click in the tmp_smz field, it set the values for the first element: rifampicina.

Expected behavior:

No repeated Ids.

I think it is because widget.uniqueId in base.xml is undefined.

**Comment by pedrobaeza** [comment]
Odoo v8 is currently not supported. Does this happen also in later versions?

**Comment by sswapnesh** [comment]
@pedrobaeza  Seems Working fine in v11.


![](https://image.prntscr.com/image/UpZ2WMrYStSfv80_WlRJwA.png)

**Comment by alesisjoan** [comment]
@sswapnesh if you click on "tmp smz" , does it set the value to "tmp smz" input or "Rifampicina" input? The misbehaving in v8 is that it's set to the first input ("Rifampicina").

**Comment by sswapnesh** [comment]
Yes, It is working fine and Inputs are Saved well.

**Comment by pedrobaeza** [comment]
Then I'm closing this as working on latest versions, but won't fix in 8.0. If you have an enterprise contract, you can put an OPW ticket to see if it's solved on that old version.

**Comment by alesisjoan** [comment]
I get some fix:

Edit Base.xml (I don't know how to inherit)

```<t t-name="FieldRadio">
    <span t-attf-class="oe_form_field oe_form_field_radio #{widget.options.horizontal ? 'oe_horizontal' : 'oe_vertical'}" t-att-style="widget.node.attrs.style">
        <span t-if="!widget.get('effective_readonly')">
            <t t-if="widget.options.horizontal">
                <t t-set="width" t-value="Math.floor(100 / widget.selection.length)"/>
                <t t-if="!widget.options.no_radiolabel">
                    <t t-foreach="widget.selection" t-as="selection">
                        <label t-att-for="widget.uniqueId + '_' + selection[0]" t-att-style="'width: ' + width + '%;'"><t t-esc="selection[1]"/></label>
                    </t>
                    <br/>
                </t>
                <t t-foreach="widget.selection" t-as="selection">
                    <div t-att-style="'width: ' + width + '%;'">
                        <span class="oe_radio_input"><input type="radio" t-att-name="widget.uniqueId" t-att-id="widget.uniqueId + '_' + selection[0]" t-att-value="selection[0]"/></span>
                    </div>
                </t>
            </t>
            <t t-if="!widget.options.horizontal">
                <t t-foreach="widget.selection" t-as="selection">
                    <div>
                        <t t-set="unico_id" t-value="_.uniqueId()"/>
                        <span class="oe_radio_input"><input type="radio" t-att-id="unico_id + '_' + selection[0]" t-att-name="unico_id" t-att-value="selection[0]"/></span><label t-if="!widget.options.no_radiolabel" t-att-for="unico_id + '_' + selection[0]"><t t-esc="selection[1]"/></label>
                    </div>
                </t>
            </t>
        </span>
        <span t-if="widget.get('effective_readonly')" class="oe_radio_readonly"><t t-esc="widget.get('value')[1]"/></span>
    </span>
</t>```

Note that the original is widget.uniqueId is undefined, so I put a variable getting the unique_id for inputs. I think it's better to modify view_form.js but don't know how to.
